[WEB-806] [turborepo] bug: npm issues on mac+ubuntu - failed to find package, MODULE_NOT_FOUND [turbo-linux-64] missing - commits passing CI on sunday now fails

### What version of Turborepo are you using?

1.8.5

### What package manager are you using / does the bug impact?

npm

### What operating system are you using?

Mac

### Describe the Bug

Since yesterday, we have made no updates to our package-lock.json, package.json nor dockerfiles, but now we get an error similar to what is described on #3328

The error occurs both locally (on e.g., OSX pcs) as well as in CI (ubuntu).

```bash
#24 60.63 npm ERR! code 1
#24 60.63 npm ERR! path /app/node_modules/turbo
#24 60.63 npm ERR! command failed
#24 60.63 npm ERR! command sh -c node install.js
#24 60.63 npm ERR! [turbo] Failed to find package "turbo-linux-64" on the file system <---- ❗️This one here changes depending on system (mac vs linux)
#24 60.63 npm ERR! 
#24 60.63 npm ERR! This can happen if you use the "--no-optional" flag. The "optionalDependencies"
#24 60.63 npm ERR! package.json feature is used by turbo to install the correct binary executable
#24 60.63 npm ERR! for your current platform. This install script will now attempt to work around
#24 60.63 npm ERR! this. If that fails, you need to remove the "--no-optional" flag to use turbo.
#24 60.63 npm ERR! 
#24 60.63 npm ERR! node:internal/modules/cjs/loader:933
#24 60.63 npm ERR!   const err = new Error(message);
#24 60.63 npm ERR!               ^
#24 60.63 npm ERR! 
#24 60.63 npm ERR! Error: Cannot find module 'turbo'
#24 60.63 npm ERR! Require stack:
#24 60.63 npm ERR! - /app/node_modules/turbo/node-platform.js
#24 60.63 npm ERR! - /app/node_modules/turbo/install.js
#24 60.63 npm ERR!     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
#24 60.63 npm ERR!     at Function.resolve (node:internal/modules/cjs/helpers:108:19)
#24 60.63 npm ERR!     at downloadedBinPath (/app/node_modules/turbo/node-platform.js:47:44)
#24 60.63 npm ERR!     at checkAndPreparePackage (/app/node_modules/turbo/install.js:[285](https://github.com/org/repo/{metadata}):15)
#24 60.63 npm ERR!     at Object.<anonymous> (/app/node_modules/turbo/install.js:310:1)
#24 60.63 npm ERR!     at Module._compile (node:internal/modules/cjs/loader:1105:14)
#24 60.63 npm ERR!     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
#24 60.63 npm ERR!     at Module.load (node:internal/modules/cjs/loader:981:32)
#24 60.63 npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
#24 60.63 npm ERR!     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12) {
#24 60.63 npm ERR!   code: 'MODULE_NOT_FOUND',
#24 60.63 npm ERR!   requireStack: [
#24 60.63 npm ERR!     '/app/node_modules/turbo/node-platform.js',
#24 60.63 npm ERR!     '/app/node_modules/turbo/install.js'
#24 60.63 npm ERR!   ]
#24 60.63 npm ERR! }
#24 60.63 
#24 60.64 npm ERR! A complete log of this run can be found in:
#24 60.64 npm ERR!     /root/.npm/_logs/2023-03-28T13_49_53_067Z-debug-0.log
#24 ERROR: process "/bin/sh -c npm ci" did not complete successfully: exit code: 1
```

I just re-ran the last passing CI (same commit hash, same lock-file, same dockerfile, cleared all our CI caches), and it now fails with the above, when it was passing yesterday.

We have tried deleting node_modules, package-lock.json and regenerating on:
- Macbook
- codespaces (ubuntu)
- windows wsl

And we keep getting issues

### Expected Behavior

Working like before :D 

### To Reproduce

Runing with `npm ci` in your with-docker instead of using yarn and:
- base image node:16.15-alpine instead of node:alpine
- turbo 1.8.5 
- npm version npm@9.6.2 instead of using yarn@1.22.17 in package.json (and replacing in dockerfiles)

The above in your example `with-docker` reproduces this on my machine (mac m2) as well

### Reproduction Repo

_No response_

<sub>[WEB-806](https://linear.app/vercel/issue/WEB-806/[turborepo]-bug-npm-issues-on-macubuntu-failed-to-find-package-module)</sub>

--- a/crates/turborepo-lockfiles/src/npm.rs
+++ b/crates/turborepo-lockfiles/src/npm.rs
@@ -24,6 +24,7 @@ pub struct NpmLockfile {
 }
 
 #[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
 struct NpmPackage {
     version: Option<String>,
     resolved: Option<String>,
@@ -321,6 +322,17 @@ mod test {
                     "node_modules/require-from-string",
                     "node_modules/table/node_modules/json-schema-traverse",
                     "node_modules/uri-js",
+                ],
+            ),
+            (
+                "node_modules/turbo",
+                vec![
+                    "node_modules/turbo-darwin-64",
+                    "node_modules/turbo-darwin-arm64",
+                    "node_modules/turbo-linux-64",
+                    "node_modules/turbo-linux-arm64",
+                    "node_modules/turbo-windows-64",
+                    "node_modules/turbo-windows-arm64",
                 ],
             ),
         ];

